### PR TITLE
Remove Field Type Mappings Manager role

### LINKED ISSUE #17991: `Field Type Mappings Manager` role does not have indices-related permissions.
## Expected Behavior
We don't know yet. May need consulting with Kay or Seth.
Possibilities:
1. Add some index-related permissions to `Field Type Mappings Manager` role (at least `read` to do CRUD on profiles, but maybe `edit` as well, to be able to set profiles on index sets).
2. Move the UI somewhere else, outside of `System->Indices`?
3. Accept that for now this new functionality is only visible to admins?
4. ...

## Current Behavior
`Field Type Mappings Manager` role does not have indices-related permissions.
Because of that, `System->Indices` page is not visible for non-admin user with this role.
Effectively, that kind of users can only work with profile via REST calls, not via GL UI.

### LINKED ISSUE #18218: Conflicting role migration for "Field Type Mappings Manager" role
## Description

I noticed I get migration messages about the "Field Type Mappings Manager" role twice on every startup.

```
2024-02-12 11:02:13,353 INFO : org.graylog2.bootstrap.ServerBootstrap - Running 112 migrations...
2024-02-12 11:02:14,066 INFO : org.graylog2.migrations.MigrationHelpers - Field Type Mappings Manager role is missing or invalid, re-adding it as a built-in role.
2024-02-12 11:02:14,108 INFO : org.graylog2.migrations.MigrationHelpers - Field Type Mappings Manager role is missing or invalid, re-adding it as a built-in role.
2024-02-12 11:02:14,135 INFO : org.graylog2.shared.initializers.PeriodicalsService - Starting 9 periodicals ...
```

The role migration for the "Field Type Mappings Manager" role is defined in two migration files:

- `V20230905081400_CreateFieldTypeMappingsManagerRole`
- `V20240118130000_FieldTypeMappingsManagerRoleReceivesProfilePermissions`

The first migration uses a different set of permissions than the second one. That's why both migrations modify the role on server startup.

We should remove one of the migrations.

## Your Environment

* Graylog Version: 6.0.0-SNAPSHOT (rev bd61c53)

### PR DESCRIPTION
**Pull request by luk-kaminski**
## Description
Remove Field Type Mappings Manager role
Fixes #17991
Fixes #18218
/nocl

## Motivation and Context
Implement Kay's decision to get rid of the role completely, and leave the related functionality for admin users only.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/migrations/RoleRemover.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    private static final Logger LOG = LoggerFactory.getLogger(RoleRemover.class);
+
+    private final MongoConnection mongoConnection;
+
+    @Inject
+    public RoleRemover(final MongoConnection mongoConnection) {
+        this.mongoConnection = mongoConnection;
+    }
+
+    public void removeBuiltinRole(final String roleName) {
```
Comment (dennisoelkers) [inline graylog2-server/src/main/java/org/graylog2/migrations/RoleRemover.java:56]: Shouldn't this be in the `MigrationHelpers` class?

Comment (luk-kaminski) [reply graylog2-server/src/main/java/org/graylog2/migrations/RoleRemover.java:56]: What's wrong with it being in a separate, dedicated class?

Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog2/migrations/RoleRemover.java:56]: For discovery: As someone who would be looking for a method to remove roles in the future, I would look in the `MigrationHelpers` class and implement a (duplicate) myself if it's not in there.

Comment (luk-kaminski) [reply graylog2-server/src/main/java/org/graylog2/migrations/RoleRemover.java:56]: I see.
I have commited a change where RoleRemover is not used directly, but via MigrationHelpers. It should solve the discovery problem.

**Pull request merged**